File: backend/src/schemas/event.py
from datetime import datetime
from typing import List, Optional

from pydantic import ConfigDict
from src.schemas.base_schema import StrictModel
from src.schemas.hall import HallResponse


class PriceResponse(StrictModel):
    id_url: str
    amount: Optional[float] = None
    available: Optional[int] = None
    expires_at: Optional[datetime] = None

    created_at: Optional[datetime] = None
    updated_at: Optional[datetime] = None

    model_config = ConfigDict(from_attributes=True)


class EventResponse(StrictModel):
    id_url: str
    production_id_url: str

    hall: Optional[HallResponse] = None

    starts_at: Optional[datetime] = None
    ends_at: Optional[datetime] = None

    order_url: Optional[str] = None

    price_urls: List[str] = []

    created_at: Optional[datetime] = None
    updated_at: Optional[datetime] = None

    model_config = ConfigDict(from_attributes=True)


class EventCreate(StrictModel):
    production_id_url: str
    hall_id_url: str
    starts_at: Optional[datetime] = None
    ends_at: Optional[datetime] = None
    order_url: Optional[str] = None


class EventUpdate(StrictModel):
    hall_id_url: Optional[str] = None
    starts_at: Optional[datetime] = None
    ends_at: Optional[datetime] = None
    order_url: Optional[str] = None

File: backend/src/schemas/event.rs
use chrono::NaiveDateTime;
use serde::{Deserialize, Serialize};

#[derive(Debug, Serialize, Clone)]
pub struct EventPriceResponse {
    pub id: i32,
    pub viernulvier_id: Option<i32>,
    pub event_id: i32,
    pub amount: Option<f64>,
    pub available: Option<bool>,
    pub expires_at: Option<NaiveDateTime>,
}

#[derive(Debug, Serialize, Clone)]
pub struct EventResponse {
    pub id: i32,
    pub viernulvier_id: Option<i32>,
    pub production_id: i32,
    pub hall_id: Option<i32>,
    pub starts_at: Option<NaiveDateTime>,
    pub ends_at: Option<NaiveDateTime>,
    pub order_url: Option<String>,
    pub created_at: Option<NaiveDateTime>,
    pub updated_at: Option<NaiveDateTime>,
    pub prices: Vec<EventPriceResponse>,
}

#[derive(Debug, Deserialize)]
pub struct EventCreate {
    pub production_id: i32,
    pub hall_id: Option<i32>,
    pub starts_at: Option<NaiveDateTime>,
    pub ends_at: Option<NaiveDateTime>,
    pub order_url: Option<String>,
}

#[derive(Debug, Deserialize)]
pub struct EventListQuery {
    pub production_id: Option<i32>,
    pub cursor: Option<i32>,
    pub limit: Option<i64>,
}
